codegen/projections/white_label/spec/endpoint_spec.rb

```diff
@@ -0,0 +1,204 @@
+          expect(endpoint.headers).to eq(expected[:headers])
+          expect(endpoint.auth_schemes).to eq(expected[:auth_schemes])
+        end
+
+        it 'produces the correct output from the client when calling endpoint_operation_with_resource' do
+          config = {}
+          config[:stub_responses] = true
+          config[:endpoint] = nil
+
+          client = Client.new(config)
+          interceptor = Hearth::Interceptor.new(read_before_transmit: proc do |context|
+            expected_uri = URI.parse(expected[:url])
+            request_uri = context.request.uri
+            expect(request_uri.hostname).to end_with(expected_uri.hostname)
+            expect(request_uri.scheme).to eq(expected_uri.scheme)
+            expect(request_uri.path).to start_with(expected_uri.path)
+            expected[:headers].each do |k,v|
+              expect(context.request.headers[k]).to eq(Hearth::HTTP::Field.new(k, v).value)
+            end
+          end)
+          client.endpoint_operation_with_resource({resource_url: 'https://resource.com/path'}, interceptors: [interceptor])
+        end
+      end
+
+      context 'Data prefix is applied' do
+        let(:expected) do
+          {
+            url: 'https://data.whitelabel.com',
+            headers: {},
+            auth_schemes: []
+          }
+        end
+
+        it 'produces the expected output from the EndpointProvider' do
+          params = Params.new(dataplane: true)
+          endpoint = subject.resolve_endpoint(params)
+          expect(endpoint.uri).to eq(expected[:url])
+          expect(endpoint.headers).to eq(expected[:headers])
+          expect(endpoint.auth_schemes).to eq(expected[:auth_schemes])
+        end
+
+        it 'produces the correct output from the client when calling dataplane_operation' do
+          config = {}
+          config[:stub_responses] = true
+          config[:endpoint] = nil
+
+          client = Client.new(config)
+          interceptor = Hearth::Interceptor.new(read_before_transmit: proc do |context|
+            expected_uri = URI.parse(expected[:url])
+            request_uri = context.request.uri
+            expect(request_uri.hostname).to end_with(expected_uri.hostname)
+            expect(request_uri.scheme).to eq(expected_uri.scheme)
+            expect(request_uri.path).to start_with(expected_uri.path)
+            expected[:headers].each do |k,v|
+              expect(context.request.headers[k]).to eq(Hearth::HTTP::Field.new(k, v).value)
+            end
+          end)
+          client.dataplane_operation({}, interceptors: [interceptor])
+        end
+
+        it 'produces the correct output from the client when calling endpoint_with_host_label_operation' do
+          config = {}
+          config[:stub_responses] = true
+          config[:endpoint] = nil
+
+          client = Client.new(config)
+          interceptor = Hearth::Interceptor.new(read_before_transmit: proc do |context|
```

Comment:
I prefer multi-step variables for cleanliness here but I'll defer to you.

Reply:
yeah, we use an intermediate elsewhere, I'll update to follow that pattern.